e2e/src/test/scala/SealedOneofSpec.scala

```diff
@@ -63,5 +63,12 @@ class SealedOneofSpec extends FlatSpec with MustMatchers {
         |}
       """.stripMargin)
   }
+
+  "message with sealed_oneof_extends" should "extend from custom user trait" in {
+    assertCompiles(
```

Comment:
Instead of `assertCompiles` it is sufficient to check if `PlayerShip.defaultInstance mustBe a [BaseTrait]` , also verify that the specific subtypes extend `BaseTrait`s. All these checks can happen at runtime rather than trigger an additional compile.